### PR TITLE
Add builds for op-deployer, bugfixes, artifacts downloads

### PR DESCRIPTION
**Pull request by mslipper**
Adds Docker builds for op-deployer, makes some bugfixes, and adds support for downloading remote artifacts.

### CODE REVIEW

File: op-chain-ops/deployer/apply.go [hunk 1]
```diff
@@ -137,28 +139,43 @@ func ApplyPipeline(
...
+			fmt.Sprintf("generate-l2-genesis-%s", chainID.Hex()),
+			func(ctx context.Context, env *pipeline.Env, artifactsFS foundry.StatDirFs, intent *state.Intent, st *state.State) error {
+				return pipeline.GenerateL2Genesis(ctx, env, artifactsFS, intent, st, chainID)
 			},
 		})
 	}
 
 	for _, stage := range pline {
-		if err := stage.stage(ctx, env, intent, st); err != nil {
+		if err := stage.stage(ctx, env, artifactsFS, intent, st); err != nil {
```
Comment (bitwiseguy) [inline op-chain-ops/deployer/apply.go]: Could `stage.stage` be renamed to `stage.execute` to make it a little more obvious what it is doing?

File: ops/docker/op-stack-go/Dockerfile [hunk 2]
```diff
@@ -101,6 +101,10 @@ ARG OP_SUPERVISOR_VERSION=v0.0.0
 RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build cd op-supervisor && make op-supervisor  \
   GOOS=$TARGETOS GOARCH=$TARGETARCH GITCOMMIT=$GIT_COMMIT GITDATE=$GIT_DATE  VERSION="$OP_SUPERVISOR_VERSION"
 
+FROM --platform=$BUILDPLATFORM builder AS op-deployer-builder
+ARG OP_NODE_VERSION=v0.0.0
+RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build cd op-chain-ops && make op-deployer  \
+  GOOS=$TARGETOS GOARCH=$TARGETARCH GITCOMMIT=$GIT_COMMIT GITDATE=$GIT_DATE VERSION="$OP_NODE_VERSION"
```
Comment (bitwiseguy) [inline ops/docker/op-stack-go/Dockerfile]: Is the `OP_NODE_VERSION` intentionally passed here or is that just a copy/paste error?

All of the args passed to the `make op-deployer` command are not actually being read or used by the `go build` command downstream. In order to use them, the makefile would need to capture them as inputs and then pass them to the `go build` command. Here is how `op-node` does it [for example](https://github.com/ethereum-optimism/optimism/blob/feat/op-deployer-6/op-node/Makefile#L17-L29)

File: op-chain-ops/deployer/init.go [hunk 3]
```diff
@@ -52,30 +67,44 @@ func Init(cfg InitConfig) error {
 	intent := &state.Intent{
 		L1ChainID:       cfg.L1ChainID,
 		UseFaultProofs:  true,
-		FundDevAccounts: cfg.Dev,
+		FundDevAccounts: true,
```
Comment (bitwiseguy) [inline op-chain-ops/deployer/init.go:70]: Do you plan to later derive the `UseFaultProofs` and `FundDevAccounts` values from input configs or will they always be `true` in this `Init()` function?

**Review by bitwiseguy — APPROVED** [review]
Looks great! Just left a few minor questions/comments

File: ops/docker/op-stack-go/Dockerfile (hunk 2 above)
Comment (mslipper) [reply ops/docker/op-stack-go/Dockerfile]: that's a copy paste error - will fix. nice catch.

File: op-chain-ops/deployer/init.go (hunk 3 above)
Comment (mslipper) [reply op-chain-ops/deployer/init.go:70]: the `Init()` function is used to generate a default config, so it'll likely always be true. users won't use the default values in production.

**Pull request merged**